components/formats-common/src/loci/common/DateTools.java

```diff
@@ -259,20 +261,10 @@ public static String formatDate(String date, String[] formats,
    * (in Unix format: milliseconds since January 1, 1970).
    */
   public static long getTime(String date, String format) {
-    final DateTimeFormatter parser =
-      DateTimeFormat.forPattern(format).withZone(DateTimeZone.UTC);
-    Instant timestamp = null;
-    try {
-      timestamp = Instant.parse(date, parser);
-    }
-    catch (IllegalArgumentException e) {
-      LOGGER.debug("Invalid timestamp '{}'", date);
-    }
-    catch (UnsupportedOperationException e) {
-      LOGGER.debug("Error parsing timestamp '{}'", date, e);
-    }
-    if (timestamp == null) return -1;
-    return timestamp.getMillis();
+    SimpleDateFormat f = new SimpleDateFormat(format);
+    Date d = f.parse(date, new ParsePosition(0));
+    if (d == null) return -1;
```

Comment:
Don't we want to continue to throw an IllegalArgumentException in this case as before, rather than returning -1, which is a valid-but-incorrect date?  There's no way to distinguish between a valid date and an error condition.